cmd/controller/main.go

```diff
@@ -73,7 +73,7 @@ func main() {
 	})
 
 	clientSet := kubernetes.NewForConfigOrDie(manager.GetConfig())
-	cloudProvider := registry.NewCloudProvider(cloudprovider.Options{ClientSet: clientSet})
+	cloudProvider := registry.NewAPI(cloudprovider.Options{ClientSet: clientSet})
```

Comment:
IMO keep this as register.NewCloudProvider. I actually renamed a bunch of factories to just cloudprovider in an earlier PR to make it easier for you.